website/content/specification/syntax/instances.md

```diff
@@ -0,0 +1,466 @@
+| [`@name`](#name) | [`token`](/specification/datatypes/#token) | required | *(no default)* |
+
+##### `@in-json`
+
+The optional `@in-json` attribute controls the representational form of a group of instances in JSON and YAML.
+
+When no attribute and value is provided for the `@in-json` attribute, the value MUST default to `SINGLETON_OR_ARRAY`.
+
+One of the following behaviors MUST be used based on the provided, or default value when no attribute and value is provided.
+
+| Value | JSON and YAML Behavior |
+|:--- |:--- |
+| [`ARRAY`](#in-jsonarray) | The child value(s) MUST be represented as an array of values. |
+| [`SINGLETON_OR_ARRAY`](#in-jsonsingleton_or_array) | If a single value is provided, then the child value MUST be that value; otherwise, for multiple values, the child values MUST be represented as an array of values. |
+| [`BY_KEY`](#in-jsonby_key) | The child value MUST be an intermediate object based on the `<json-key>`.
+
+###### `@in-json="ARRAY"`
+
+TODO: P2: complete this section.
+###### `@in-json="SINGLETON_OR_ARRAY"`
+
+TODO: P2: complete this section.
+###### `@in-json="BY_KEY"`
+
+
+TODO: P2: Address issue https://github.com/usnistgov/metaschema/issues/316
+TODO: P2: complete this section using the text below.
+
+When used in this way, the property names of this intermediate object will be the value of the flag as specified by the `@json-key` attribute on the definition referenced by the `@ref` on the instance. The value of the intermediate object property will be an object or value , with property names equal to the value of the referenced `define-field` or `define-assembly` component's flag as specified by the `@json-key` attribute on that component.
+
+One problem with zero-or-more cardinality as supported by `fields` and `assemblies` is that in JSON, no suitable structure is available for the inclusion of truly arbitrary but repeatable properties or 'contents' (as to its structural type) on an object. The closest thing is an array, which can be pulled into use for this -- at the cost of not permitting a JSON property label on items in the array. In order to capture the same information as is transparently available on the XML, it is therefore necessary to 'finesse' the JSON object type: Metaschema does this by mapping each field or assembly in a zero-or-many set, to an array with the corresponding number of items. The name of the objects can thus be captured implicitly, by naming (labeling) their containing array.
+
+This works, but there are also occasions when a much more concise mapping may also be supported -- if the data can be ensured to follow another rule, namely that data elements (string data) can be known to be uniquely-valued. In these cases there is a different option, namely to promote a flag of a particular known (and controlled) type, to a role as "address" -- which can (incidentally) serve as a label on a JSON property, thus improving both presentation, and addressability.
```

Comment:
Doesn't this reflect an old design since superseded? as we now have `<json-key-flag>`... maybe want to mark this as in need of revision. No `field/@address` appears in the Metaschema schema currently (for example).

Reply:
Indeed. I haven't reworked this.